### PR TITLE
Diagonal smoothing change

### PR DESCRIPTION
**Pull request by xxalpha**
Changed it so underlays for walls is handled in the smoothing code instead of ChangeTurf.

Also changed how underlays are picked, from previous turf appearance to adjacent turf appearance. Until walls are objects, this is the cleaner way to do this.

### CODE REVIEW

**Comment by duncathan** [comment]
Walls will never be objects

**Comment by xxalpha** [comment]
Unbeliever!

File: code/__HELPERS/icon_smoothing.dm [hunk 1]
```diff
@@ -98,34 +98,60 @@
 			var/adjacencies = calculate_adjacencies(A)
 
 			if(A.smooth & SMOOTH_DIAGONAL)
-				diagonal_smooth(A, adjacencies)
+				A.diagonal_smooth(adjacencies)
 			else
 				cardinal_smooth(A, adjacencies)
 
-/proc/diagonal_smooth(atom/A, adjacencies) //TODO: atom smoothing procs, optimize diagonal per previous state & underlay generation
+/atom/proc/diagonal_smooth(adjacencies)
```
Comment (Iamgoofball) [inline code/__HELPERS/icon_smoothing.dm]: what if i want to smooth datums

HUH?????

Comment (xxalpha) [inline code/__HELPERS/icon_smoothing.dm]: Any datum that is supposed to have a physical presence in the ``world`` needs to be a subtype of /atom.

How long have you been coding for byond and you still don't know this.

Comment (Iamgoofball) [inline code/__HELPERS/icon_smoothing.dm]: :joy:

Comment (PKPenguin321) [inline code/__HELPERS/icon_smoothing.dm]: 👌💯

**Pull request merged**